client/src/global_state_query.rs

```diff
@@ -1,71 +1,85 @@
-        "Identifies the hash value provided as either a block hash or state root hash. You must provide either a block hash or a state root hash";
-    const BLOCK: &str = "block";
-    const STATE_ROOT: &str = "state";
+    pub(super) const ARG_NAME: &str = "state-root-hash";
+    const ARG_SHORT: &str = "s";
+    const ARG_VALUE_NAME: &str = ARG_HEX_STRING;
+    const ARG_HELP: &str = "Hex-encoded hash of the state root";
 
     pub(super) fn arg() -> Arg<'static, 'static> {
         Arg::with_name(ARG_NAME)
-            .required(true)
-            .possible_value(BLOCK)
-            .possible_value(STATE_ROOT)
+            .long(ARG_NAME)
+            .short(ARG_SHORT)
+            .required(false)
             .value_name(ARG_VALUE_NAME)
             .help(ARG_HELP)
-            .display_order(DisplayOrder::StateIdentifier as usize)
+            .display_order(DisplayOrder::StateRootHash as usize)
     }
 
-    pub(super) fn get<'a>(matches: &'a ArgMatches) -> &'a str {
-        matches
-            .value_of(ARG_NAME)
-            .unwrap_or_else(|| panic!("should have {} arg", ARG_NAME))
+    pub fn get<'a>(matches: &'a ArgMatches) -> Option<&'a str> {
+        matches.value_of(ARG_NAME)
     }
 }
 
-/// Handles providing the arg for and retrieval of either the block hash or state root hash.
-mod hash {
+mod block_hash {
     use super::*;
 
-    const ARG_NAME: &str = "hash";
-    const ARG_VALUE_NAME: &str = "HEX STRING";
-    const ARG_HELP: &str = "Hex-encoded block hash or state root hash";
+    pub(super) const ARG_NAME: &str = "block-hash";
+    const ARG_SHORT: &str = "b";
+    const ARG_VALUE_NAME: &str = ARG_HEX_STRING;
+    const ARG_HELP: &str = "Hex-encoded hash of the block";
 
     pub(super) fn arg() -> Arg<'static, 'static> {
         Arg::with_name(ARG_NAME)
-            .required(true)
+            .long(ARG_NAME)
+            .short(ARG_SHORT)
+            .required(false)
             .value_name(ARG_VALUE_NAME)
             .help(ARG_HELP)
-            .display_order(DisplayOrder::Hash as usize)
+            .display_order(DisplayOrder::BlockHash as usize)
     }
 
-    pub(super) fn get<'a>(matches: &'a ArgMatches) -> &'a str {
-        matches
-            .value_of(ARG_NAME)
-            .unwrap_or_else(|| panic!("should have {} arg", ARG_NAME))
+    pub fn get<'a>(matches: &'a ArgMatches) -> Option<&'a str> {
+        matches.value_of(ARG_NAME)
+    }
+}
+
+fn global_state_str_params<'a>(matches: &'a ArgMatches) -> GlobalStateStrParams<'a> {
+    if let Some(state_root_hash) = state_root_hash::get(matches) {
+        return GlobalStateStrParams {
+            is_block_hash: false,
+            hash_value: state_root_hash,
+        };
+    }
+    if let Some(block_hash) = block_hash::get(matches) {
+        return GlobalStateStrParams {
+            is_block_hash: true,
+            hash_value: block_hash,
+        };
     }
+    unreachable!("clap arg groups and parsing should prevent this")
```

Comment:
Could we change this message to include the function name `global_state_str_params`? That way if this `unreachable!` is ever reached, someone will have a nice and easy time grepping for it.

Reply:
Nice point